andela/ah-scorpion
# #159726511 Add feature for users to be able to view comment edit history

#### What does this PR do?
- allows users to edit their comments
- allows users to see their comment edit history

#### Description of tasks performed
- added a CommentHistory model and created migrations
- added a CommentHistory Serializer
- added URL for comment history
- added GenericView methods for updating comments and getting comment update history `views.py` module
- Added unit tests for the feature

# How can this PR be manually tested?
**Note**: Use Postman to make requests
- use git pull to get the source code from ` ft-comment-edit-history-159726511 ` branch
- use `source .env` to export environmental variables and activate virtual env
- run `python manage.py` migrate to apply recent migrations
- run `python manage.py runserver` to start the project
- if not already done, create a new user account and login
- if not already done, create an article and copy the slug
- if not already done, create a comment for the above article and in the json response received, copy the `id`
- make the a `POST` request using this url: `/api/v1/articles/<slug>/comments/<comment_id>/` replacing the <slug> with the slug you copied above and <comment_id> with the id you copied above.
Include the following `json` request 
```
{
    "content": "<your new comment here>"
}
```
- if you used a valid id above, you should get a response which looks like this:
```
{
    "content": "<your new comment here>",
    "parent": <parent id of the comment. null if the comment has no parent>
}
```
- if the comment id is not valid you should get this response:
```
{
    "message": "Comment not found"
}
```
- if the new comment is the same as existing, you should get this response:
```
{
    "message": "New comment same as the existing. Editing rejected"
}
```
- make a `GET` request to the following url: `/api/v1/articles/<slug>/comments/history/<comment_id>/` replacing the <slug> with the slug you copied above and <comment_id> with the id you copied above.
- you should get a list of all edits made to the comment. The response whould look like this:
```
[
    {
        "id": <id>,
        "comment": "<this is the initial comment>",
        "date_created": "<time when edit was made>",
        "parent_comment": <id of the column edited>
    },
    {
        "id": <id>,
        "comment": "<this is the new comment>",
        "date_created": "<time when edit was made>",
        "parent_comment": <id of the column edited>
    }
]
```

#### What are the related PT Stories
- [#159726511](https://www.pivotaltracker.com/story/show/159726511)
